### PR TITLE
fix: explorative analytics ui issues

### PR DESCRIPTION
**Pull request by tyyzqmf**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

1. **Full window view** for dashboard
2. Retention select data display error
3. Dropdown selection of events to display the **platform** 
4. Patch attribute data type to display name if attributes has the same name

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [x] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: frontend/src/pages/pipelines/comps/BasicInfo.tsx [hunk 1]
```diff
@@ -48,8 +48,8 @@ const BasicInfo: React.FC<BasicInfoProps> = (props: BasicInfoProps) => {
     setLoadingRetry(true);
     try {
       const resData: ApiResponse<null> = await retryPipeline({
-        pid: pipelineInfo?.projectId || '',
-        id: pipelineInfo?.pipelineId || '',
+        pid: pipelineInfo?.projectId ?? '',
```
Comment (chenhaiyun) [inline frontend/src/pages/pipelines/comps/BasicInfo.tsx:51]: use `defaultStr(xxxx)`

File: frontend/src/pages/pipelines/comps/BasicInfo.tsx [hunk 2]
```diff
@@ -65,8 +65,8 @@ const BasicInfo: React.FC<BasicInfoProps> = (props: BasicInfoProps) => {
     setLoadingUpgrade(true);
     try {
       const resData: ApiResponse<null> = await upgradePipeline({
-        pid: pipelineInfo?.projectId || '',
-        id: pipelineInfo?.pipelineId || '',
+        pid: pipelineInfo?.projectId ?? '',
```
Comment (chenhaiyun) [inline frontend/src/pages/pipelines/comps/BasicInfo.tsx:68]: use `defaultStr(xxxx)`

File: frontend/src/pages/pipelines/comps/BasicInfo.tsx [hunk 3]
```diff
@@ -179,7 +179,7 @@ const BasicInfo: React.FC<BasicInfoProps> = (props: BasicInfoProps) => {
               <Box variant="awsui-key-label">
                 {t('pipeline:detail.version')}
               </Box>
-              <div>{pipelineInfo?.templateInfo?.pipelineVersion || '-'}</div>
+              <div>{pipelineInfo?.templateInfo?.pipelineVersion ?? '-'}</div>
```
Comment (chenhaiyun) [inline frontend/src/pages/pipelines/comps/BasicInfo.tsx:182]: use `defaultStr(xxxx, "-")`

File: frontend/src/pages/pipelines/comps/BasicInfo.tsx [hunk 4]
```diff
@@ -207,8 +207,8 @@ const BasicInfo: React.FC<BasicInfoProps> = (props: BasicInfoProps) => {
               <Link
                 external
                 href={buildVPCLink(
-                  pipelineInfo?.region || '',
-                  pipelineInfo?.network.vpcId || ''
+                  pipelineInfo?.region ?? '',
```
Comment (chenhaiyun) [inline frontend/src/pages/pipelines/comps/BasicInfo.tsx:210]: use `defaultStr(xxxx)`

File: frontend/src/pages/pipelines/comps/BasicInfo.tsx [hunk 5]
```diff
@@ -221,8 +221,8 @@ const BasicInfo: React.FC<BasicInfoProps> = (props: BasicInfoProps) => {
               <Link
                 external
                 href={buildS3Link(
-                  pipelineInfo?.region || '',
-                  pipelineInfo?.bucket.name || '',
+                  pipelineInfo?.region ?? '',
```
Comment (chenhaiyun) [inline frontend/src/pages/pipelines/comps/BasicInfo.tsx:224]: use `defaultStr(xxxx)`

File: frontend/src/pages/analytics/retention/AnalyticsRetention.tsx [hunk 6]
```diff
@@ -535,11 +535,11 @@ const AnalyticsRetention: React.FC<AnalyticsRetentionProps> = (
                   });
                 }}
                 changeStartEvent={(index, item) => {
-                  const eventName = item?.value;
+                  const eventName = item?.name;
                   const eventParameters = getEventParameters(eventName);
                   const parameterOption = parametersConvertToCategoryItemType(
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/retention/AnalyticsRetention.tsx:540]: Consider set the second parameter of the `parametersConvertToCategoryItemType` with default value like `[]`, there are too many `eventParameters ?? []`.

File: frontend/src/pages/pipelines/comps/BasicInfo.tsx (hunk 2 above)
Comment (tyyzqmf) [reply frontend/src/pages/pipelines/comps/BasicInfo.tsx:68]: fix all defaultStr

File: frontend/src/pages/analytics/retention/AnalyticsRetention.tsx (hunk 6 above)
Comment (tyyzqmf) [reply frontend/src/pages/analytics/retention/AnalyticsRetention.tsx:540]: fixed.

**Review by zxkane — APPROVED** [review]

**Review by chenhaiyun — APPROVED** [review]

**Pull request merged**